plugins/gdk-pixbuf/pixbufloader-jxl.c

```diff
@@ -501,8 +471,10 @@ static gboolean load_increment(gpointer context, const guchar *buf, guint size,
                         "Failed to allocate output pixel buffer");
             return FALSE;
           }
+          gdk_pixbuf_set_option(frame.data, "icc-profile", decoder_state->icc_base64);
           decoder_state->pixel_format.align =
               gdk_pixbuf_get_rowstride(frame.data);
+          decoder_state->pixel_format.data_type = JXL_TYPE_UINT8;
```

Comment:
GDK pixbuf only supports 8-bit. I think it might be worthwhile to request an HDR space and attach an HDR icc profile if the image is tagged as HDR (e.g. PQ)